### PR TITLE
[RFR] Prevent the move wrong tile error

### PR DESCRIPTION
**Pull request by Luwangel**
- [x] Catch the api error and do nothing after a not possible move

### CODE REVIEW

File: app/src/Api/GameApi.php [hunk 1]
```diff
@@ -26,13 +28,18 @@ public function new(int $size) : Game {
   }
 
   public function move(Game $game, int $tile) : Game {
```
Comment (jdemangeon) [inline app/src/Api/GameApi.php:30]: Do you have a corresponding test?

File: app/src/Api/GameApi.php [hunk 2]
```diff
@@ -26,13 +28,18 @@ public function new(int $size) : Game {
...
+    try {
+      $response = $this->client->post('/move-tile', [
+        'body' => json_encode([
+          'InitialGrid' => $game->getResolvedGrid(),
+          'Grid' => $game->getCurrentGrid(),
+          'TileNumber' => $tile
+        ])
+      ]);
+    } catch (ConnectException $e) {
+      return $game;
```
Comment (alexisjanvier) [inline app/src/Api/GameApi.php:40]: so the user doesn't see that there was an error?

Comment (alexisjanvier) [reply app/src/Api/GameApi.php:40]: IMO, you should display something like "You can't move that tile. Try another one".

**Review by alexisjanvier — CHANGES_REQUESTED** [review]
Missing a test and error message

**Review by alexisjanvier — APPROVED** [review]

**Pull request merged**